habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py

```diff
@@ -293,11 +299,16 @@ def act(
                     rnn_hidden_states,
                     prev_actions,
                 )
-                rnn_hidden_states[batch_ids] *= 0.0
-                prev_actions[batch_ids] *= 0
-            self._cur_skills = (
-                (~self._call_high_level) * self._cur_skills
-            ) + (self._call_high_level * new_skills)
+                if "rnn_hidden_states" not in hl_info:
+                    rnn_hidden_states[batch_ids] *= 0.0
+                    prev_actions[batch_ids] *= 0
+                elif self._skills[skill_id].num_recurrent_layers != 0:
+                    raise ValueError(
+                        f"The code does not currently support neural LL and neural HL skills. Skill={self._skills[skill_id]}, HL={self._high_level_policy}"
```

Comment:
Hard to understand why the skill having non zero number of recurrent layers means that the user is trying to use neural HL and neural LL. Add a comment of give an property to skills with a descriptive name that checks if the LL is neural.
Also, aren't we trying to have frozen NN LL skills and training NN HL policy ? This error seems to indicate this is not supported.

Reply:
Correct, that combination is currently not supported. I refactored with a more descriptive property.